components/script/dom/document.rs

```diff
@@ -1862,6 +1864,108 @@ impl Document {
+                        }),
+                        &global,
+                    )
+                    .unwrap();
+            },
+            script_traits::GamepadEvent::Disconnected(index) => {
+                // 2. Queue a task on the gamepad task source to perform the following steps:
+                let document = Trusted::new(self);
+                self.window()
+                    .task_manager()
+                    .gamepad_task_source()
+                    .queue(
+                        task!(gamepad_disconnected: move || {
+                            let document = document.root();
+                            let window = document.window();
+
+                            // 2.1 Set gamepad.[[connected]] to false.
+                            // 2.2 Let document be gamepad's relevant global object's associated Document;
+                            //     otherwise null.
+                            // 2.3 If gamepad.[[exposed]] is true and document is not null and is fully active,
+                            //     then fire an event named gamepaddisconnected at gamepad's relevant global object
+                            //     using GamepadEvent with its gamepad attribute initialized to gamepad.
+                            // 2.4 Let navigator be gamepad's relevant global object's Navigator object.
+                            // 2.5 Set navigator.[[gamepads]][gamepad.index] to null.
+                            // 2.6 While navigator.[[gamepads]] is not empty and the last item of \
+                            //     navigator.[[gamepads]] is null, remove the last item of navigator.[[gamepads]].
+                            let gamepad_list = window.Navigator().GetGamepads();
+                            if gamepad_list.Length() > 0 {
+                                gamepad_list.remove_gamepad(index.0);
+                            }
+                        }),
+                        &global,
+                    )
+                    .unwrap();
+            },
+            // https://www.w3.org/TR/gamepad/#receiving-inputs
+            script_traits::GamepadEvent::Updated(index, update_type) => {
+                let document = Trusted::new(self);
+                // 2. Queue a task on the gamepad task source to update gamepad state for gamepad.
+                self.window().task_manager().gamepad_task_source().queue(
+                    task!(gamepad_updated: move || {
+                        let document = document.root();
+                        let window = document.window();
+                        let gamepad_list = window.Navigator().GetGamepads();
+                        if let Some(gamepad) = gamepad_list.IndexedGetter(index.0 as u32) {
+                            // 1. Let now be the current high resolution time.
+                            let current_time = time::get_time();
+                            let now = (current_time.sec * 1000 + current_time.nsec as i64 / 1000000) as f64;
```

Comment:
we are trying to remove `time` create, can you use std::time? 

cc @mrobinson

Reply:
Yes, done.